pkg/controller/clusterclaim/clusterclaim_controller.go

```diff
@@ -48,14 +56,87 @@ func AddToManager(mgr manager.Manager, r *ReconcileClusterClaim) error {
 	}
 
 	// Watch for changes to ClusterClaim
-	err = c.Watch(&source.Kind{Type: &hivev1.ClusterClaim{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
+	if err := c.Watch(&source.Kind{Type: &hivev1.ClusterClaim{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return err
+	}
+
+	// Watch for changes to ClusterDeployment
+	if err := c.Watch(
+		&source.Kind{Type: &hivev1.ClusterDeployment{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(requestsForClusterDeployment),
+		},
+	); err != nil {
+		return err
+	}
+
+	// Watch for changes to the hive-claim-owner Role
+	if err := c.Watch(
+		&source.Kind{Type: &rbacv1.Role{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: requestsForRBACResources(r.Client, hiveClaimOwnerRoleName, r.logger),
+		},
+	); err != nil {
+		return err
+	}
+
+	// Watch for changes to the hive-claim-owner RoleBinding
+	if err := c.Watch(
+		&source.Kind{Type: &rbacv1.Role{}},
```

Comment:
```suggestion
		&source.Kind{Type: &rbacv1.RoleBinding{}},
```

copypasta? See #2024